sympy/functions/elementary/piecewise.py

```diff
@@ -806,6 +807,25 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         args = [(ec.expr._eval_nseries(x, n, logx), ec.cond) for ec in self.args]
         return self.func(*args)
 
+    def _eval_factor(self):
+        from sympy.polys import factor, cancel
+        from sympy.polys.polyerrors import PolynomialError
+
+        args = [(e if hasattr(e, "_eval_factor") else factor(e), c) for e, c in self.args]
+        commonfact = None
+        for expr, cond in args:
+            if commonfact is not None:
+                try:
+                    commonfact = gcd(commonfact, expr)
+                except PolynomialError:
+                    commonfact = 1
+            else:
+                commonfact = expr
+        if commonfact == 1 and args == self.args:
+            return self
+        args = [(cancel(e/commonfact), c) for e, c in args]
+        return commonfact*Piecewise(*args)
```

Comment:
I think that you can use `evaluate=False` for `Piecewise` here to speed things up as there should be no need to reevaluate the arguments. (Unless there is an instance of the same expression which cannot be identified upon the first creation...)